nano/node/rpc.cpp

```diff
@@ -941,7 +966,30 @@ void nano::rpc_handler::blocks_info ()
 						block->serialize_json (contents);
 						entry.put ("contents", contents);
 					}
-
+					// Subtype check
+					if (block->type () == nano::block_type::state)
+					{
```

Comment:
This block (20+ lines) is identical to that in `block_info ()` can it be refactored into a separate function?